packages/flutter/test/material/flat_button_test.dart

```diff
@@ -333,6 +333,79 @@ void main() {
         paintsExactlyCountTimes(#clipPath, 0),
     );
   });
+
+  testWidgets('FlatButton responds to tap and onLongPress when enabled', (WidgetTester tester) async {
+
+    int pressedCount = 0;
+
+    Widget buildFrame({VoidCallback onPressed, VoidCallback onLongPress}) {
+      return Directionality(
+        textDirection: TextDirection.ltr,
+        child: FlatButton(child: const Text('button'), onPressed: onPressed, onLongPress: onLongPress),
+      );
+    }
+
+    // onPressed not null, onLongPress null.
+    await tester.pumpWidget(
+      buildFrame(onPressed: () { pressedCount += 1; }, onLongPress: null),
+    );
+    expect(tester.widget<FlatButton>(find.byType(FlatButton)).enabled, true);
+    await tester.tap(find.byType(FlatButton));
+    await tester.pumpAndSettle();
+    expect(pressedCount, 1);
+
+    // onPressed null, onLongPress not null.
+    pressedCount = 0;
+    await tester.pumpWidget(
+      buildFrame(onPressed: null, onLongPress: () { pressedCount += 1; }),
+    );
+    expect(tester.widget<FlatButton>(find.byType(FlatButton)).enabled, true);
+    await tester.longPress(find.byType(FlatButton));
+    await tester.pumpAndSettle();
+    expect(pressedCount, 1);
+
+    // onPressed null, onLongPress null.
+    pressedCount = 0;
+    await tester.pumpWidget(
+      buildFrame(onPressed: null, onLongPress: null),
+    );
+    expect(tester.widget<FlatButton>(find.byType(FlatButton)).enabled, false);
+    await tester.tap(find.byType(FlatButton));
+    await tester.longPress(find.byType(FlatButton));
+    await tester.pumpAndSettle();
+    expect(pressedCount, 0);
+  });
+
+  testWidgets('FlatButton onPressed and onLongPress callbacks are distincly recognized', (WidgetTester tester) async {
+    bool didPressButton = false;
+    bool didLongPressButton = false;
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: FlatButton(
+          onPressed: () {
+            didPressButton = true;
+          },
+          onLongPress: () {
+            didLongPressButton = true;
+          },
+          child: const Text('button'),
+        ),
+      ),
+    );
+
+    final Finder flatButton = find.byType(FlatButton);
```

Comment:
This is much easier to read 👍

Reply:
Agree.. But there is a problem. I'm redefining the button 3 times in the other tests. So I would need to either use flatbutton1, flatbutton2, flatbutton.. or a var, which also doesn't sound good.

Ideas?

Reply:
You could remove the `final` declaration and re-use the same variable, which is fine